### PR TITLE
multi-proc test execution

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Takes advantage of new MSBuild by enabling parallel execution of tests. On my 8-proc machine, reduced time to execute tests from 10 mins to 6.5 mins.

/cc @brthor @eerhardt @livarcocc @rainersigwald

### CODE REVIEW

File: build/Microsoft.DotNet.Cli.Test.targets [hunk 1]
```diff
@@ -27,13 +27,28 @@
...
+          EnvironmentVariables=$(RunTestEnvironmentVariables);
+          TestProjectName=%(TestProjects.OutputName);
+          TestResultXmlDir=$(TestResultXmlDir);
+          ToolPath=$(Stage2Directory);
+          WorkingDirectory=%(TestProjects.ProjectDir)
+        </Properties>
+      </ProjectsToTest>
+    </ItemGroup>
+
+    <MSBuild
```
Comment (livarcocc) [inline build/Microsoft.DotNet.Cli.Test.targets]: How does this replaces invoking dotnet test? Just curious.

Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Test.targets]: look at the batched <MSBuild target just below.

They way that one works is that invokes the default target [or specified] on the project that it points to. Each batch has the option to override Properties inside of that proj. What these are doing are creating one Item for each test project and defining the appropriate properties to make it run. They then invoke this new test project which does the actual invocation of dotnet-test. By batching on an MSBuild task we're telling MSBuild it's ok to treat each of these projects as independent, which allows it to run them on multiple cores!

**Comment by livarcocc** [comment]
Now that you have all the files, :shipit:

File: build/Microsoft.DotNet.Cli.Compile.targets [hunk 2]
```diff
@@ -161,7 +161,7 @@
             DestinationFiles="$(SdkOutputDirectory)/$(HostPolicyBaseName)" />
 
       <!-- copy core sdk -->
```
Comment (eerhardt) [inline build/Microsoft.DotNet.Cli.Compile.targets]: Looks like this comment is out of date

**Comment by eerhardt** [comment]
Looks good to me. Assuming it is green

**Comment by eerhardt** [comment]
Can we do the same for "BuildTests"?  It would seem the tests should be able to be built in a multi-proc fashion.

**Comment by piotrpMSFT** [comment]
In phases, yes. The build itself is straightforward, but getting the projects to work in parallel is time consuming because failures are less deterministic.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test  Ubuntu16.04 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please.

**Pull request merged**